torch/_meta_registrations.py

```diff
@@ -6178,6 +6178,34 @@ def meta_searchsorted(
         return torch.empty((), dtype=dtype, device=sorted_sequence.device)
 
 
+def _check_for_unsupported_isin_dtype(dtype):
+    torch._check(
+        not any(
+            dtype == ud
+            for ud in [torch.bool, torch.bfloat16, torch.complex128, torch.complex64]
+        ),
+        lambda: f"Unsupported input type encountered for isin(): {dtype}",
+    )
+
+
+@register_meta(aten.isin)
+@out_wrapper()
+def meta_isin(elements, test_elements, *, assume_unique=False, invert=False):
```

Comment:
Why did you choose to write a separate meta function instead of adding these dtype checks to the decomposition? Were there other failures as well?